bag/interface/database.py

```diff
@@ -578,7 +576,7 @@ def _import_design(self, lib_name, cell_name, imported_cells, dsn_db, new_lib_pa
 
         # update netlist file
         content = self.parse_schematic_template(lib_name, cell_name)
-        sch_info = yaml.load(content, Loader=yaml.FullLoader)
+        sch_info = read_yaml(content)
```

Comment:
This line breaks our device generation. Apparently, at BWRC, they expect to receive a filename from the parse_schematic_template function, which would then be read in using read_yaml. However, observing the output from the Skill interface (using BWRC master), one will find that it is actually the netlist info in yaml format (rather than a filename), which needs to parsed using yaml.load (see corresponding line in our fork). In essence, reverting this file back to the state of our master branch fixes the generation proceudre, in my limited testing at least.

To my understanding, there is a possibility that the skill interface will return a filename instead of the netlist info. This will happen when the output size exceeds the Virtuoso variable size limitation (further details in bag/interface/skill.py). However, in bag/interface/zmqwrapper.py the BAG server response handler (see function recv_obj starting from line 89 of the file), will ALWAYS return a Python object (dictionary) describing the netlist info.

Either we investigate further, or revert this file back to the state of our master branch.

Reply:
read_yaml is defined in bag/io/file.py . The method just contains the yaml.load command, but I think the function id defined in weird manner

`def read_yaml(fname):
    """Read the given file using YAML.

    Parameters
    ----------
    fname : string
        the file name.

    Returns
    -------
    content : Any
        the object returned by YAML.
    """
    with open_file(fname, 'r') as f:
        content = yaml.load(f, Loader=yaml.Loader)

    return content
`
'with open' opens the file for reading and content stored to f which is then parsed with load, which should take filename as an argument. I suggest that we make a fix with separete commit, and then propose that as a fix to upstream. I try to figure out a way to pint the person who made this change. 'git blame <filename> is handy :) .

Reply:
fixed

Reply:
Thanks Ayan. @sporrasm , I think we are ready to merge.